[Lockdown] Update list of trusted fonts for Facebook.com

vitorroriz opened the pull request:
#### 0cd843152f93498d6b35765dba84fa2771b30478
<pre>
[Lockdown] Update list of trusted fonts for Facebook.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=304598">https://bugs.webkit.org/show_bug.cgi?id=304598</a>
<a href="https://rdar.apple.com/161923840">rdar://161923840</a>

Reviewed by Matthieu Dubet.

Safe font parser is not available on down levels so we need
to update the list of trusted fonts for Facebook.com

* Source/WebCore/loader/cache/TrustedFonts.cpp:
(WebCore::trustedFontHashesInLockdownMode):

Canonical link: <a href="https://commits.webkit.org/304866@main">https://commits.webkit.org/304866@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3f124a16fd389a1af8e0de898fa58ca02a5a4225

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136711 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/159/builds/9087 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47998 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144439 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/89684 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a2f319a7-2a6f-4b53-b4a3-f42c6b941d9b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138583 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9771 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8916 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104543 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76331 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6b78280d-0473-40c6-b05a-9bf934296c3c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139656 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7143 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122498 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85382 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/22bb5e4f-5f9b-49f8-9b15-cb44cfd86687) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/6787 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4476 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5031 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/116107 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40685 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147196 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8754 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41258 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112896 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8772 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/7366 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113225 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28761 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6705 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118788 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/62905 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8802 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36842 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8523 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72368 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8742 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8594 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mdubet commented on Source/WebCore/loader/cache/TrustedFonts.cpp:
Should we add a comment to say which font it is (like the previous ones) ?

vitorroriz replied:
I think it is unnecessary as these names are just some identifiers defined by fb with no value for us.

Andrew-Fryer commented on Source/WebCore/loader/cache/TrustedFonts.cpp:
```diff
@@ -854,7 +854,18 @@ static const MemoryCompactLookupOnlyRobinHoodHashSet<AtomString>& trustedFontHas
+        "4lwRbTGm3tahH1/N4+/Y2X4K+yngJLAUyDzhf4q8kOQ="_s,
+        "WY/G2MZa1/vsJr+lZc0AAsVBwuxuLxJEE0zdu5z5jSk="_s,
+        "OKhOTmvrlR2wFVV65RzWwZrDm01wTM76amx7a8yybyM="_s,
+        "mXkRKROZST2m7meFkdxZmUGDQ5Q5IxgaInsfPAAoAbY="_s });
```
Out of curiosity, why do we label some with the web font file name, but not others?

vitorroriz replied:
I've started labeling it as documentation, which is maybe useful whey they use a real name like "font-awesome-whatever" but these identifiers don't really help us here. I end up having to compute the base64(sha256(font)) to know if they are equivalent anyway.

vitorroriz committed:
[Lockdown] Update list of trusted fonts for Facebook.com
https://bugs.webkit.org/show_bug.cgi?id=304598
rdar://161923840

Reviewed by Matthieu Dubet.

Safe font parser is not available on down levels so we need
to update the list of trusted fonts for Facebook.com

* Source/WebCore/loader/cache/TrustedFonts.cpp:
(WebCore::trustedFontHashesInLockdownMode):

Canonical link: https://commits.webkit.org/304866@main
